src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -103,29 +103,28 @@ class RsExtractFunctionConfig private constructor(
             if (wherePredList.isEmpty()) return ""
             val typeParams = typeParameters().map { it.declaredType }
             if (typeParams.isEmpty()) return ""
-            val filtered = wherePredList.filter { typeParams.contains(it.typeReference?.type) }
+            val filtered = wherePredList.filter { it.typeReference?.type in typeParams }
             if (filtered.isEmpty()) return ""
             return filtered.joinToString(separator = ",", prefix = " where ") { it.text }
         }
 
     private fun typeParameters(): List<RsTypeParameter> {
         val bounds = typeParameterBounds()
-        val paramAndReturnTypes = (parameters.mapNotNull { it.type } + listOf(returnValue).mapNotNull { it?.type })
-        return containingFunction.typeParameters.filter {
-            val typeParameter = it.declaredType
-            paramAndReturnTypes.any {
-                it.types().contains(typeParameter) || it.dependTypes(bounds).contains(typeParameter)
-            }
+        val paramAndReturnTypes = mutableSetOf<Ty>()
+        (parameters.mapNotNull { it.type } + listOfNotNull(returnValue?.type)).forEach {
+            paramAndReturnTypes.addAll(it.types())
+            paramAndReturnTypes.addAll(it.dependTypes(bounds))
         }
+        return containingFunction.typeParameters.filter { it.declaredType in paramAndReturnTypes }
     }
 
-    private fun typeParameterBounds(): Map<Ty, List<Ty>> {
+    private fun typeParameterBounds(): Map<Ty, Set<Ty>> {
         return containingFunction.typeParameters.associate {
             val type = it.declaredType
             val bounds = it.bounds.flatMap {
                 it.bound.traitRef?.path?.typeArgumentList?.typeReferenceList?.flatMap { it.type.types() } ?: emptyList()
             }
-            type to bounds
+            type to bounds.toSet()
```

Comment:
You can use `flatMapTo(HashSet<Ty>())` instead of `flatMap` above to put results into set immediately and avoid this transformation here

Reply:
use flatMapTo?

Reply:
Oops, my mistake
yep, I've meant `flatMapTo`